Add: Zombie Flags Report Documentation (#530)
configcat/docs: website/docs/my-zombie-flags-report.mdx

---
id: my-zombie-flags-report
title: My Zombie Flags Report
description: This page explains how to use The Zombie Flags Report to list all feature flags that weren't flipped for a while.
---

Zombie flags are feature flags that are no longer needed or rarely used. Over time, they contribute to technical debt and, if unmanaged, may cause unintended errors or issues.

Here are a few ways zombie flags can form:

> - A flag was initially tied to a feature that has since become permanent and is no longer controlled by the flag.

Reviewer: ```suggestion
- A feature flag originally controlled a feature that has since become permanent, and the flag is no longer needed. 
```

- A decision was made to keep a feature on or off, yet the flag still lingers in your code or DevOps workflow.

- The flag was used for a one-time deployment, but the rollout is over.

- You haven't toggled the flag in a long time.

## Managing Zombie Flags

The Zombie Flags (Stale Flags) Report lists all feature flags that weren't flipped for a while. There is a high chance these flags are not needed anymore but are still referenced in your source code, and ConfigCat recommends removing these flags to avoid tech debt.

### Setting Up the Zombie Feature Flags Report

> Login to your [ConfigCat dashboard](https://app.configcat.com), click your Organization name at the top right, then select **My Zombie Flags Report**.

Reviewer 2: I think it would be better to link directly to the page: https://app.configcat.com/my-account/zombie-flags-report

<img alt="Navigating to My Zombie Flags Report" src="/docs/assets/guide/my-zombie-flags-report/navigating-to-my-zombie-flags-report_192dpi.png" width="1224" decoding="async" loading="lazy" />

Here is what the **My Zombie Flags Report** page looks like:

<img alt="My Zombie Flags Report" src="/docs/assets/guide/my-zombie-flags-report/my-zombie-flags-report_192dpi.png" width="1366" decoding="async" loading="lazy" />

> Here, you could enable the **Receive report via email** option and select how often the report should be sent by setting the **Frequency**. The **Criteria** option sets how long a flag should be inactive before it is considered stale. Use the **Scope** option to select which flags should be in the report. You can decide if you want the report to include all your feature flags or only the ones you are explicitly watching. Lastly, select the **environment**.

Reviewer 2: I think the "Lastly, select the **environment**." part is missing what you will happen when you select the environments.

description: This page explains how to use The Zombie Flags Report to list all feature flags that weren't flipped for a while.
---

Zombie flags, also known as stale feature flags, are feature flags that are no longer needed or rarely used. Over time, they contribute to technical debt and, if left unmanaged, may cause unintended errors, performance issues, or bloated codebases. 

## What Are Zombie (Stale) Feature Flags?

Zombie flags typically form when: 

- A feature flag originally controlled a feature that has since become permanent, and the flag is no longer needed. 

- A decision was made to keep a feature on or off, yet the flag still lingers in your code or DevOps workflow.

- The flag was used for a one-time deployment or experiment that is now complete. 

- The flag hasn't been toggled or referenced in a long time. 

## Why Should You Remove Zombie Flags?

Keeping unused feature flags in your code can:

- Increase technical complexity and reduce maintainability.

- Lead to unexpected behavior if old flags are triggered.

- Waste time during code reviews or onboarding.

- Negatively impact performance in large-scale applications. 

## Managing Zombie Flags

> The Zombie Flags (Stale Flags) Report helps you identify feature flags that haven't been updated or toggled recently. These flags are most likely obsolete but still referenced in your source code. 

Reviewer: ```suggestion
My Zombie Flags Report scans your ConfigCat environments and identifies feature flags that haven’t been updated or toggled within a specified timeframe. It highlights flags that are likely obsolete but still present in your source code or workflows.
```

> The Zombie Flags (Stale Flags) Report helps you identify feature flags that haven't been updated or flipped for a while. These flags are most likely obsolete but still referenced in your source code. 

Reviewer: ```suggestion
The Zombie Flags (Stale Flags) Report helps you identify feature flags that haven't been updated or toggled recently. These flags are most likely obsolete but still referenced in your source code. 
```

### How to Set Up Zombie Feature Flags Report

1. Log in to your [ConfigCat dashboard](https://app.configcat.com/my-account/zombie-flags-report).
2. Click your Organization name in the top-right corner.
3. Select  **My Zombie Flags Report** from the dropdown menu. 

<img alt="Navigating to My Zombie Flags Report" src="/docs/assets/guide/my-zombie-flags-report/navigating-to-my-zombie-flags-report_192dpi.png" width="1224" decoding="async" loading="lazy" />

### Exploring the My Zombie Flags Report Page

On the My Zombie Flags Report page, you can customize the report to fit your workflow:

- **Receive Report via Email:** Get regular email updates with stale flag info. 
- **Frequency:** Set how often you'd like to receive the report (e.g., weekly or monthly).
- **Criteria:** Define how long a flag must be inactive before it's marked as stale.
- **Scope:** Choose to include all feature flags or only the ones you're watching. 

> - Environment: Select the ConfigCat [**environment**](https://configcat.com/docs/main-concepts/#environment) for which you want to analyze flags. 

Reviewer: ```suggestion
- **Environment:** Select the ConfigCat [**environment**](https://configcat.com/docs/main-concepts/#environment) for which you want to analyze flags. 
```

> - **Environment:** Select the ConfigCat [**environment**](https://configcat.com/docs/main-concepts/#environment) for which you want to analyze flags. 

Reviewer: ```suggestion
- **Environment:** Select the ConfigCat [environment](https://configcat.com/docs/main-concepts/#environment) for which you want to analyze flags. 
```

<img alt="My Zombie Flags Report" src="/docs/assets/guide/my-zombie-flags-report/my-zombie-flags-report_192dpi.png" width="1366" decoding="async" loading="lazy" />

## Sample Email Report

> Once your preferences are set, ConfigCat will email you a Zombie Flags Report. Here's a preview of what that report may look like: 

Reviewer: ```suggestion
Once your preferences are set, ConfigCat will email you a Zombie Flags Report. Here's a preview of what that report might look like: 
```

<img alt="Zombie Flags Report Email" src="/docs/assets/guide/my-zombie-flags-report/zombie-flags-report-email_192dpi.png" width="1366" decoding="async" loading="lazy" />

## Other Tools for Identifying Unused Feature Flags

> In addition to the Zombie Flags Report, we offer other tools to help you clean up unused flags: 

Reviewer: ```suggestion
In addition to the My Zombie Flags Report, ConfigCat provides several tools to help you detect and clean up unused feature flags:
```
